sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -121,17 +120,19 @@ private boolean isDecimalTypeMatched(DataType dt) {
 
   private boolean canReadAsIntDecimal(DataType dt) {
     if (!DecimalType.is32BitDecimalType(dt)) return false;
-    return isDecimalTypeMatched(dt);
+    DecimalType d = (DecimalType) dt;
+    return isDecimalTypeMatched(d) || d.scale() == 0;
```

Comment:
`!isDecimalTypeMatched(d)` doesn't mean `DecimalMetadata` is null. Shall we put the `|| d.scale() == 0` in `isDecimalTypeMatched`?